### PR TITLE
[FLINK-16125][kafka] Remove unnecessary zookeeper.connect property validation

### PR DESCRIPTION
**Pull request by PatrickRen**
## What is the purpose of the change

This pull request removes the unnecessary validation of zookeeper.connect for kafka connectors, since it is not needed after kafka connector version 0.8.

## Brief change log
- Remove validation of zookeeper.connect property

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by wuchong** [comment]
Thanks for the contribution @PatrickRen . Could you also remove this line in the IT case?https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableTestBase.java#L86

Btw, could you also open a PR for release-1.10 ? For 1.10,  I think we can't just remove this property, because we still provide `Kafka08TableSourceSinkFactory` in 1.10. We can make this property optional for 0.9+, but required for 0.8 in 1.10.

**Comment by PatrickRen** [comment]
Thank you for the review @wuchong ! Sure I'll fix it and create another PR for release-1.10 later.

**Comment by PatrickRen** [comment]
Hi Jark @wuchong , I noticed that if we remove the line you mentioned, the variable `zk` will become unused:
https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableTestBase.java#L69

But it is still referenced in part of commented code: 
https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableTestBase.java#L115

Should I remove the variable `zk`, or keep it in case of those commented code being used in the future? I'm not sure about the purpose of the commented code...

**Comment by PatrickRen** [comment]
Any comment from Becket @becketqin ?

**Comment by becketqin** [comment]
@PatrickRen I did a quick code search, and found a bunch of "zookeeper.connect" usages in the project, most, if not all, of them are for Kafka. Can we clean them up in this patch as well?

**Comment by PatrickRen** [comment]
Sure @becketqin , I'll check and clean them up soon. Thanks for your review!

**Comment by wuchong** [comment]
> Hi Jark @wuchong , I noticed that if we remove the line you mentioned, the variable `zk` will become unused:
> https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableTestBase.java#L69
> 
> But it is still referenced in part of commented code:
> https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTableTestBase.java#L115
> 
> Should I remove the variable `zk`, or keep it in case of those commented code being used in the future? I'm not sure about the purpose of the commented code...

Yes, you are right. We should remove them too.

**Comment by PatrickRen** [comment]
@flinkbot run travis
@flinkbot run azure

File: flink-end-to-end-tests/flink-confluent-schema-registry/src/main/java/org/apache/flink/schema/registry/test/TestAvroConsumerConfluent.java [hunk 1]
```diff
@@ -49,14 +49,12 @@ public static void main(String[] args) throws Exception {
 			System.out.println("Missing parameters!\n" +
 				"Usage: Kafka --input-topic <topic> --output-string-topic <topic> --output-avro-topic <topic> " +
 				"--bootstrap.servers <kafka brokers> " +
-				"--zookeeper.connect <zk quorum> " +
```
Comment (wuchong) [inline flink-end-to-end-tests/flink-confluent-schema-registry/src/main/java/org/apache/flink/schema/registry/test/TestAvroConsumerConfluent.java]: Hi @becketqin , could you help to review this? 
This PR also remove `zookeeper.connect` properties for non-table-modules.

**Comment by PatrickRen** [comment]
@flinkbot run travis
@flinkbot run azure

**Comment by PatrickRen** [comment]
@flinkbot run travis
@flinkbot run azure

**Comment by PatrickRen** [comment]
@flinkbot run travis
@flinkbot run azure

**Comment by PatrickRen** [comment]
Hi Becket @becketqin and Jark @wuchong , can you help to review the code? I think it's in good shape now. Travis and Azure tests have passed. Thank you~

**Comment by becketqin** [comment]
@PatrickRen Thanks for the patch. LGTM.
merged to master 
f93346c93c6a841cdce576d48a0b5ca8076cc195

**Pull request closed without merging**